### PR TITLE
Docs for `requireLogin`

### LINKED ISSUE #4509: Documentation for requireLogin Class
Part of #4132.
Add documentation for the new requireLogin class so that we can add that to the new buttons etc in future as per Jeff's suggestion https://github.com/publiclab/plots2/issues/4132#issuecomment-451003321.

### PR DESCRIPTION
**Pull request by oorjitchowdhary**
Fixes #4509 

I have added the documentation for the `requireLogin` class. Renamed the file to `LOGIN_SYSTEMS.md` and also fixed grammatical errors in the document.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] PR is descriptively titled 📑
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

File: doc/LOGIN_SYSTEMS.md [hunk 1]
```diff
@@ -56,6 +59,14 @@ Or, write them in config/application.yml file
 Add the app_id and app_secret in the Jenkins and containers/docker*.yml files in the production.
 They are accessed by ENV["OAUTH_GITHUB_APP_KEY"], ENV["OAUTH_GITHUB_APP_SECRET"] etc inside the (config/initializers/omniauth.rb)[https://github.com/publiclab/plots2/blob/master/config/initializers/omniauth.rb]
 
+## How to setup login on various locations?
+
+For better UI on the website, we recently created login and signup modals. The corresponding code is present here https://github.com/publiclab/plots2/blob/master/app/views/layouts/_header.html.erb#L176-L266.
```
Comment (SidharthBansal) [inline doc/LOGIN_SYSTEMS.md]: We don't write amigious words like here in the documentations. please remove it.

File: doc/LOGIN_SYSTEMS.md [hunk 2]
```diff
@@ -56,6 +59,14 @@ Or, write them in config/application.yml file
...
+
+The `/login` and `/signup` pages still exist as some particular features require the pages.
```
Comment (SidharthBansal) [inline doc/LOGIN_SYSTEMS.md]: Some is yet another ambiguous term

File: doc/LOGIN_SYSTEMS.md [hunk 3]
```diff
@@ -56,6 +59,14 @@ Or, write them in config/application.yml file
...
+
+A custom Javascript class named as `requireLogin` is active and it can be inserted in anywhere for asking the user to log in first to use that particular function. For example, it can be implemented in a button with the button class as `btn btn-default requireLogin`.
```
Comment (SidharthBansal) [inline doc/LOGIN_SYSTEMS.md]: Please simplify this to a single line to have better readability.
All the documentations are for active functions. So active keyword is redundant.

File: doc/LOGIN_SYSTEMS.md [hunk 4]
```diff
@@ -1,22 +1,25 @@
...
-For a new account creation, ``email_prefix`` i.e. the part of email before ``@`` symbol, is used as username. In case there exists a user with the same username then randomly generated hexadecimal code is appended to the email_prefix. Then this email_prefix is used as username.
+* Github: Using the [omniauth-github](https://github.com/omniauth/omniauth-github) gem for OAuth.
+* Google: Using the [omniauth-google-oauth2](https://github.com/zquestz/omniauth-google-oauth2) gem for OAuth.
+* Twitter: Using the [omniauth-twitter](https://github.com/arunagw/omniauth-twitter) gem for OAuth.
+* Facebook: Using the [omniauth-facebook](https://github.com/mkdynamic/omniauth-facebook) gem for OAuth.
+
+To sign up or log in via a provider or to link a provider, go to https://publiclab.org/auth/:provider. Sign in through the provider with the desired email address and password. Then the user will be redirected to https://publiclab.org/auth/:provider/callback. Authentication hash is available in the callback. It can be accessed by `request.env['omniauth.auth']` for any provider. Now the user signs up, logs in or links his/her providers' accounts to his/her Public Lab account.
```
Comment (SidharthBansal) [inline doc/LOGIN_SYSTEMS.md]: Please change his/her to their in this documentation. Sorry I missed it.

File: doc/LOGIN_SYSTEMS.md (hunk 3 above)
Comment (Dsxv) [inline doc/LOGIN_SYSTEMS.md]: @oorjitchowdhary  @SidharthBansal  this single doc has solved many of my doubts as things that seems obvious for you guys are not so obvious for new comers that's why Thankyou so much!! :D

**Comment by SidharthBansal** [comment]
I created this. Thanks Pawan.

**Comment by oorjitchowdhary** [comment]
@SidharthBansal I've made the changes

**Comment by oorjitchowdhary** [comment]
@SidharthBansal Is it good now?

**Comment by oorjitchowdhary** [comment]
@SidharthBansal done

File: doc/LOGIN_SYSTEMS.md (hunk 3 above)
Comment (SidharthBansal) [reply doc/LOGIN_SYSTEMS.md]: Thanks I made it in GSoC'19 with help of @jywarren

**Comment by SidharthBansal** [comment]
Thanks @oorjitchowdhary .
I hope you learnt characteristics of good documentation here.

**Pull request merged**